.github/workflows/release.yml

```diff
@@ -0,0 +1,100 @@
+name: release
+
+on:
+  push:
+    tags: 
+      - '*.*.*'
+
+jobs:
+  
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - name: Install poetry
+        run: |
+          curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+          echo "PATH=$HOME/.poetry/bin:$PATH" >> $GITHUB_ENV
+      - name: Build distributions
+        run: |
+          poetry build
+      - name: Upload distribution artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: perceval-dist
+          path: dist
+  
+  tests:
+    needs: [build]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Download distribution artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: perceval-dist
+          path: dist
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dev dependencies
+        run: |
+          pip install httpretty>=0.9.6
```

Comment:
Didn't you find any workaround for this?

Reply:
I couldn't install only dev dependencies since the feature is not yet released in poetry.

I can use `pip install requirements_tests.txt` but we are looking forward to removing them soon, so I didn't consider the option.
WDYT?

Reply:
I'm not sure if we should keep this PR open until we fix the too problems we have with poetry now:

- Install scripts
- Install dev dependencies only

For the first, there's a PR open that fixes the problem but it isn't integrated with the core of the code yet. For the second one, the fix was implemented but it's not available yet in any of the releases of poetry. Am I right @vchrombie ? Maybe you can complete this with the links to the PRs.

What do you think? Should we wait and see, or mege this?

Reply:
Yes, you are right @sduenas.

>     * Install scripts

There is an existing PR that could fix this issue https://github.com/python-poetry/poetry-core/pull/40. I see some active work going on here. So, I'm expecting this to completed soon and we could make use of this feature.

>     * Install dev dependencies only

This request was proposed long back and the PR https://github.com/python-poetry/poetry/pull/2917 was merged too, but it is not available for any release till now. I'm not sure when it would be available to use.

> What do you think? Should we wait and see, or merge this?

We are more concerned with the first problem as we cannot use the `perceval` command if we install the tool using poetry. I think once the feature PR is merged, I can open a new PR by adding the configuration.

We currently have a workaround for the second problem (installing the dev dependencies manually) as we don't have another option.

I think we should go ahead and merge this now. We can update the workflows once we have these features in the poetry releases.

WDYT?

Reply:
Ok to me. What should we do with the perceval script then?

Reply:
> Ok to me. What should we do with the perceval script then?

If we want to generate a release with this workflow, it can cause problems for the users if they are using the pip packages. This is not safe and not recommended.

This means we should not generate the tags until we get the perceval script working with poetry.
Can we generate the releases without the tags (if we merge this PR)? If this is not possible, then we shouldn't merge this at the moment and leave it till we fix the script issue.

@sduenas, do you have any thoughts?